### PR TITLE
Migrate go-versions CI to GitHub Actions 

### PR DESCRIPTION
**Pull request by nikita-bykov**
In scope of this PR we've moved go-versions CI from AzureDevOps to GitHub Actions

Details:
The main goal of these changes is to switch the repository to use the system account to publish Go releases and create PRs with updating the `versions-manifest.json file`. For this purpose we use `GITHUB_TOKEN `which is automatically generated by GitHub. 

Unfortunately, we were not able to get rid of the personal access token completely. We have to authenticate using a personal access token to trigger workflow configured with the `workflow_dispatch` event.

**GitHub Actions CI approach:**
We've configured two workflows:

- `build-go-packages.yml` - to build, test and publish Go packages
- `create-pr.yml` - to create PRs with updating the versions-manifest.json file

All workflows are configured using the workflow_dispatch event and can be started manually.

We did not move get-go-versions pipeline, it is still on the AzureDevOps

Issue: https://github.com/actions/virtual-environments-internal/issues/921

### CODE REVIEW

File: .github/workflows/build-go-packages.yml [hunk 1]
```diff
@@ -0,0 +1,167 @@
...
+        Invoke-Pester -Script ./Go.Tests.ps1 -EnableExit
+      working-directory: ./tests
+
+  publish_release:
+    name: Publish release
+    if: github.event.inputs.PUBLISH_RELEASES == 'true'
+    needs: test_go
+    runs-on: ubuntu-latest
+    outputs:
+      upload_url: ${{ steps.create_release.outputs.upload_url }}
```
Comment (MaksimZhukov) [inline .github/workflows/build-go-packages.yml:119]: We don't need this section

File: .github/workflows/build-go-packages.yml [hunk 2]
```diff
@@ -0,0 +1,167 @@
...
+            });
+          }
+
+  trigger_pr:
+    name: Trigger "Create Pull Request" workflow
+    needs: publish_release
+    runs-on: ubuntu-latest
+    steps:
+    - name: Trigger "Create Pull Request" workflow
+      uses: actions/github-script@v2
```
Comment (MaksimZhukov) [inline .github/workflows/build-go-packages.yml:158]: Does it work as expected? I think we should use `v3` of this action

File: tests/Go.Tests.ps1 [hunk 3]
```diff
@@ -1,27 +1,30 @@
...
-    return $useGoLogFile.Fullname
+        $useGoLogFile = Get-ChildItem -Path $logsFolderPath | Where-Object {
+            $logContent = Get-Content $_.Fullname -Raw
+            return $logContent -match "setup-go@v"
+        } | Select-Object -First 1
+        return $useGoLogFile.Fullname
+    }
 }
 
+
```
Comment (MaksimZhukov) [inline tests/Go.Tests.ps1:20]: Extra paragraph

File: .github/workflows/build-go-packages.yml (hunk 1 above)
Comment (nikita-bykov) [reply .github/workflows/build-go-packages.yml]: Removed it

File: .github/workflows/build-go-packages.yml (hunk 2 above)
Comment (nikita-bykov) [reply .github/workflows/build-go-packages.yml]: Changed the version

File: tests/Go.Tests.ps1 (hunk 3 above)
Comment (nikita-bykov) [reply tests/Go.Tests.ps1]: Removed it

**Review by maxim-lobanov — APPROVED** [review]

**Review by MaksimZhukov — APPROVED** [review]

**Review by AlenaSviridenko — APPROVED** [review]

**Pull request merged**